flamegpu/introduction.rst

```diff
@@ -3,44 +3,43 @@
-but provides a natural and flexible environment for studying systems
+The focus on individuals is considerably more computationally demanding
+yet provides a natural and flexible environment for studying systems
 demonstrating emergent behaviour. Despite the obvious parallelism,
 traditionally frameworks for ABM fail to exploit this and are often
 based on highly serialised algorithms for manipulating mobile discrete
 agents. Such an approach has serious implications, placing stringent
 limitations on both the scale of models and the speed at which they may
 be simulated. The purpose of the FLAME GPU framework is to address the
-limitations of previous agent modelling software by targeting the high
+limitations of previous agent modelling software by targeting the high-
 performance GPU architecture. The framework is designed with parallelism
 in mind and as such allows agent models to scale to massive sizes and
-ensures simulations run within reasonable time constrains. In addition
-to this visualisation is easily achievable as simulation data is held
+ensures simulations run within reasonable time constrains. 
+Furthermore, visualisation is easily achievable as simulation data is held
 entirely within GPU memory where it can be rendered directly.
 
-High Level Overview of FLAME GPU
+High-Level Overview of FLAME GPU
 --------------------------------
 
-Technically the FLAME GPU framework is not a simulator, it is instead a
-template based simulation environment that maps formal descriptions of
+Technically, the FLAME GPU framework is not a simulator, it is instead a
+template-based simulation environment that maps formal descriptions of
 agents into simulation code. The representation of an agent is based on
 the concept of a communicating X-Machine (which is an extension to the
 Finite State Machine which includes memory). Whilst the X-Machine has
-very formal definition X-Machine agents can be thought of a state
-machines which are able to communicate via messages which are stored in
-a globally accessible message lists. Agent functionality is exposed as a
+very formal definition X-Machine agents can be thought of as state
+machines able to communicate via messages stored in
+globally accessible message lists. Agent functionality is exposed as a
 set of state transition functions which move agents from one internal
 state to another. Upon changing state, agents update their internal
 memory through the influence of messages which may be either used as
 input (by iterating message lists) or as output (where information may
 be passed to the message lists for other agents to read). FLAME GPU uses
-agent function scripting for this purpose where script is defined in a
-number of *Agent Function Files*. Simulation models are specified using
+agent function scripting for this purpose where the script is defined via an *Agent Function File*. Simulation models are specified using
```

Comment:
Using multiple files is supported, so I'd rather this line were reverted to the previous version.

Reply:
Sorry for the lateness in replying, I got carried away by other activities. I was meant to incorporate your suggestion and get back to you with the following.
My previous understanding about the single file limitation was based on [2.2.2 Function Files](https://flamegpu.readthedocs.io/en/master/flamegpu/modelspec.html#function-files) quoted here,

> The `functionFiles` element is not optional and must contain a single file element which defines the name of a source code file which holds the scripted agent functions  

Based on your comment, I would say that paragraph is outdated. Therefore, I propose to fix it as follows,  
> The `functionFiles` element is not optional and must contain ***at least*** a single file element which defines the name of a source code file which holds the scripted agent functions

If this seems correct, I can send another pull request with this change.

Reply:
Your suggestion seems good to me.